packages/dm-core-plugins/src/job/JobPlugin.tsx

```diff
@@ -90,86 +87,34 @@ export const JobPlugin = (props: IUIPlugin & { config: JobPluginConfig }) => {
     applicationInput: {
       type: EBlueprint.REFERENCE,
       referenceType: 'link',
-      address: jobInputAddress(),
+      address: jobInputAddress,
     },
     runner: config?.runner,
   }
 
   if (config?.outputTarget) jobEntity.outputTarget = config.outputTarget
 
-  const updateDocument = async (
-    jobAddress: string,
-    jobEntityFormData: TJob
-  ): Promise<unknown> => {
-    return DmssApi.documentUpdate({
-      idAddress: jobAddress,
-      data: JSON.stringify(jobEntityFormData),
-    })
-      .then(() => {
-        setJobEntityId(jobAddress)
-        return Promise.resolve()
-      })
-      .catch((error: AxiosError<ErrorResponse>) => {
-        console.error(error.response?.data)
-      })
-  }
-
-  const addDocument = async (
-    jobAddress: string,
-    jobEntityFormData: TJob
-  ): Promise<unknown> => {
+  const addDocument = async (): Promise<unknown> => {
     return DmssApi.documentAdd({
-      address: jobAddress,
-      document: JSON.stringify(jobEntityFormData),
+      address: jobTargetAddress,
+      document: JSON.stringify(jobEntity),
+    }).catch((error: AxiosError<ErrorResponse>) => {
+      console.error(error.response?.data)
     })
-      .then((res) => {
-        // The UID cannot be used as ID before the job has been started.
-        // Also, the uid returned from the addDocument endpoint differs
-        // from the one returned from the startJob endpoint.
-        // setJobEntityId(`${dataSourceId}/$${response.data.uid}`)
-        setJobEntityId(jobAddress)
-        return Promise.resolve(res)
-      })
-      .catch((error: AxiosError<ErrorResponse>) => {
-        console.error(error.response?.data)
-      })
   }
 
-  useEffect(() => {
-    if (jobEntityId.length > 0 && allowStartJob) {
-      start().then((res) => {
-        setJobId(res?.uid)
-        setJobExists(true)
-      })
-    }
-  }, [jobEntityId])
-
   function createNewJob(): Promise<unknown> {
-    setAllowJobStart(true)
     if (jobExists) {
-      return updateDocument(jobTargetAddress(), jobEntity)
+      return updateDocument(jobEntity, false).then(() => setJobExists(true))
```

Comment:
Do we need to setJobExistst -> true here?  It is already inside the `if...` because `jobExists`